### DISCUSSION TITLE
c3 super mini adaption for version 2

### DISCUSSION THREAD

**Discussion by dtashmore547** [mp-se/gravitymon#190] [General]
I have updated my super mini boards to the current dev version and everything is working fine, I still have a desire to see the led in operation as it gives a useful indication of the state of the device.
I am lost as to why the led should interfere with the gyro as I have the gyro on gpio6 and gpio7 whereas the blue on-board LED is on gpio8. I would love to see the LED operating the same as the on-board LED of the esp8266 mini. is this possible? could I define the super mini as a separate device place this definition in all the places where the esp32c3 definition is used except the led.hpp and led.cpp and place the gpio8 as LED_INBUILT in the main.hpp folder or would that cause conflicts?

**Comment by mp-se** [comment]
On the esp32 c3 mini from wemos the LED is tied to PIN 7 so thats where the conflict comes from. If you board uses a different GPIO then the led can be used but that would require a specialized target to be added to support both boards.

**Comment by mp-se** [comment]
what is the board defintion you are using in arduino ?

**Reply by dtashmore547** [reply]
hi
this is the board info I get from arduino.
![IMG_20240904_092718](https://github.com/user-attachments/assets/dc0d2b65-78cd-453a-b9bc-e06444a8ce46)
I gotta this helps. all the info I can find on the esp32c3 super mini claims the inbuilt led is on gpio8

**Reply by mp-se** [reply]
Doesent really help, do you have a link to the manufacturer?

**Comment by dtashmore547** [comment]
this is the link to the item I am using, there are a lot of these devices on aliexpress, have you never seen them before, they are very small and compact.
ZAR27.60 | TENSTAR ROBOT ESP32-C3 ESP32-S3 ESP32-H2 ESP32-C6 Development Board ESP32 C3 SuperMini WiFi Bluetooth ESP32C3
https://a.aliexpress.com/_EQ8niPn

**Comment by mp-se** [comment]
This board is not yet supported in the arduino for esp32. Its on the todo-list which means it will be included in arduino 3.x which gravitymon don't use yet.  So we just have to wait and see when this board is supported on how to proceed.

**Reply by dtashmore547** [reply]
This surprises me as it is just the basic esp32C3 chip on a small form factor board there is very little added to it, as far as i can make out the only real difference is that the rgb LED is replaced with a single blue led. i would expect the device to work exactly as the other board with the exception of the LED. my build so far is working fine with the exception of the LED.

**Reply by mp-se** [reply]
Well, it there are changes then there needs to be a separate board definition file, if there is a similar board definition then most of the features will work. In your case everything expect the led since the pin for controlling that is different. It's possible to patch the framework if one creates a board definition.

**Comment by dtashmore547** [comment]
that is what I thought, I shall see what I can do, I was just unsure if I would be conflicting with something else.
thanks

**Reply by mp-se** [reply]
You can start with changing the LED_PIN in the arduino_pins.h for your board, should be in the framework_dir / variants / board_name / pins_arduino.h.

I've just added a script in the dev branch that patches the definition for the s3 mini since that config is wrong and they only fixed that in 3.x

**Reply by simonidelh** [reply]
@mp-se: I've done a port to the Dev-KIT C V4 via your defines. Do you want to merge this port in your release? It's basically a Floaty, so no DS implemented. But maybe interesting for others or @dtashmore547

**Reply by mp-se** [reply]
The dev boards are not really useful for doing a build. I dont want to merge anything into the main project unless its possible to build a compete working device from it.

**Reply by simonidelh** [reply]
well, with the gyro, you get a working device. But understood, you don't want the board in main

**Reply by mp-se** [reply]
Thats not what i wrote, ill be happy to add it if there is a tested and verified device that can be supported. If you have build instructions and can show a working hardware setup i can add it to main.

**Comment by dtashmore547** [comment]
hi Magnus
I still have to get back to my c3 build, I was struggling to get the LED to work (just lack of knowledge) it is a blue LED and it is supposed to be on the 8 pin of the super mini module, i just need to work out how to get it defined correctly in your code. everything else is working so far but I miss the LED indication for when it is in configuration mode. I hope to get back to it soon and get my device complete with dedicated circuit board and onboard wireless charging circuit in a finalised condition, if i succeed i will contact you, so far it works with no LED indicator. 
regards and thanks for your amazing work.

On 26 December 2024 23:37:18 GMT+02:00, Magnus ***@***.***> wrote:
>Thats not what i wrote, ill be happy to add it if there is a tested and verified device that can be supported. If you have build instructions and can show a working hardware setup i can add it to main. 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/discussions/190#discussioncomment-11671293
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

**Reply by mp-se** [reply]
what is the board definition you are using ? I can have a look at how its declaration.

**Comment by dtashmore547** [comment]
hi Magnus
I found a board which was similar on the web but i couldn't get it to compile when i defined it.
the board only had esp32c3 super mini on it, was cheap on aliexpress, looked the ideal size to fit into the housings available, definitely fits better than the regular board into the rapt pill housings that i am using.
i am busy making my board available on pcbway, it's not perfect but works (switching from protel to kicad wasn't easy for me after 15 years working with protel until i retired) the component footprint for the super mini is .5mm too wide and i forgot to make the holes larger so the pins don't fit in, to compensate i stuck the device down with a 3m double sided pad and wired the connections which worked fine. if i can get it working 100 percent i can update the pcb. the operation is good so far but i still have to get the balancing done in the rapt pill housing, the wireless charging is working well, it is very useful with the rapt pill housing as that is hard to open and close with the double o-ring housing.
in the info i can find about it says the led of which there are two, 1 on the supply which i crippled as it would only be a battery drain, the other on pin 8 is a blue led and i feel it is the only thing holding me back. hope I haven't gone on too much i just feel it may be a useful addition to your stable.
regards
Dave

On 27 December 2024 09:58:58 GMT+02:00, Magnus ***@***.***> wrote:
>what is the board definition you are using ? I can have a look at how its declaration. 
>
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/discussions/190#discussioncomment-11675726
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

**Reply by mp-se** [reply]
It looks like this board does not have the RBG led that the standard C3 has so that might be the issue. My framework assumes that a C3 board uses the RBG led. But this is something that can be fixed quite easily. This part is in my common espframework project in the led.cpp/hpp. just remove the defines for the C3 so it does not use the neopixelWrite(...) method and it should work. I can add an option to disable the RBG code in the framework.

The easy way is to download the source and but that under lib/espfwk and then remove the dependecy from the platformio.ini file. That way you should be able to get past that step.

**Reply by mp-se** [reply]
I just recalled that I had issues with the C3 mini board and the LED. On that board the LED was also the pin that was connected to the gyro so I had to disable the use of the LED on that board. You can check if thats the case for your board as well.

**Reply by mp-se** [reply]
I have updated my espframework so it checks if the board has RGB led or not. You can also disable the led function if needed. Gravitymon is also updated (dev branch) to use the latest version and I have updated most of the dependant libraries to the latest versions.

**Comment by dtashmore547** [comment]
that's very helpful thanks
DTAshmore

**Comment by dtashmore547** [comment]
hi Magnus
finally got back to my C3 build, i can confirm that unlike the C3 D1 mini which has the RGB led on pin 7, my C3 super mini board has a single blue led on pin 8, i have tested the operation of this led and it is fine, if i enable the led function in the platform.ini file the gyro creases to work, i tried to define the led_builtin to pin 8 but this doesn't work, i am unsure how to proceed. everything else is working great on my board but i would still like to get the led working, i am not sure how to patch the esp framework. 
regards
Dave

On 28 December 2024 17:22:39 GMT+02:00, Magnus ***@***.***> wrote:
>I just recalled that I had issues with the C3 mini board and the LED. On that board the LED was also the pin that was connected to the gyro so I had to disable the use of the LED on that board. You can check if thats the case for your board as well. 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/discussions/190#discussioncomment-11685453
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

**Reply by mp-se** [reply]
You cannot set the LED_PIN in your project this needs to be set in Arduino board configuration, typically the pins_arduino.h file for that board. You only set the board type in platformio.ini which then defines the pins. If you board is not part of the supported board lists you need to create those defintions yourself in a fork of the arduino framework. 

An simpler option is to use one of the board defintions and "patch" the pins_arduino.h before a build. I have done that for the s3 board since the defines for the RGB was faulty. 

See; script/patch_arduino.py for an example.

**Comment by dtashmore547** [comment]
hi Magnus
that clarifies a lot for me, i found this page on Google which i shall try. 
<https://sigmdel.ca/michel/ha/esp8266/super_mini_esp32c3_en.html>
looks promising

On 1 January 2025 13:53:19 GMT+02:00, Magnus ***@***.***> wrote:
>You cannot set the LED_PIN in your project this needs to be set in Arduino board configuration, typically the pins_arduino.h file for that board. You only set the board type in platformio.ini which then defines the pins. If you board is not part of the supported board lists you need to create those defintions yourself in a fork of the arduino framework. 
>
>An simpler option is to use one of the board defintions and "patch" the pins_arduino.h before a build. I have done that for the s3 board since the defines for the RGB was faulty. 
>
>See; script/patch_arduino.py for an example. 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/discussions/190#discussioncomment-11709796
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

**Comment by dtashmore547** [comment]
hi Magnus
i have success with those instructions, downloaded the 2 files and placed as instructed, didn't work so copied the pins_arduino to the same directory as the esp32-hal-gpio.h and the esp32c3 board compiled, removed the -D ESPFWK_DISABLE_LED flag and the led is flashing. 
thanks for your help, no idea how this can be incorporated into your code but i will help where i can. 
1 possible problem i noticed it's that when i first logged onto the configuration page there was no gyro detected, i unplugged and plugged back into the usb and it came right, is this a possible problem?
Dave

On 1 January 2025 13:53:19 GMT+02:00, Magnus ***@***.***> wrote:
>You cannot set the LED_PIN in your project this needs to be set in Arduino board configuration, typically the pins_arduino.h file for that board. You only set the board type in platformio.ini which then defines the pins. If you board is not part of the supported board lists you need to create those defintions yourself in a fork of the arduino framework. 
>
>An simpler option is to use one of the board defintions and "patch" the pins_arduino.h before a build. I have done that for the s3 board since the defines for the RGB was faulty. 
>
>See; script/patch_arduino.py for an example. 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/discussions/190#discussioncomment-11709796
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

**Reply by mp-se** [reply]
Missing gyro is some bug in the i2c driver or hardware after flashing, require a reboot to work properly.

**Comment by dtashmore547** [comment]
hi Magnus
i have been very happy with the results of my changes to get my espc3 super-mini working, should i pass on my changes to you or post them on the gravitymon forum post? so far i have a board on pcbway which is a little imperfect but completely useable, i intend to update it in time. it is possible to stick the board onto the standard ispindel board with an M3 double sided pad and fly wire it as it is such a small board. i can send the 2 files with instructions along with the small changes to the platform.ini file if you wish.
regards
Dave

On 1 January 2025 17:28:17 GMT+02:00, ***@***.*** wrote:
>hi Magnus
>i have success with those instructions, downloaded the 2 files and placed as instructed, didn't work so copied the pins_arduino to the same directory as the esp32-hal-gpio.h and the esp32c3 board compiled, removed the -D ESPFWK_DISABLE_LED flag and the led is flashing. 
>thanks for your help, no idea how this can be incorporated into your code but i will help where i can. 
>1 possible problem i noticed it's that when i first logged onto the configuration page there was no gyro detected, i unplugged and plugged back into the usb and it came right, is this a possible problem?
>Dave
>
>On 1 January 2025 13:53:19 GMT+02:00, Magnus ***@***.***> wrote:
>>You cannot set the LED_PIN in your project this needs to be set in Arduino board configuration, typically the pins_arduino.h file for that board. You only set the board type in platformio.ini which then defines the pins. If you board is not part of the supported board lists you need to create those defintions yourself in a fork of the arduino framework. 
>>
>>An simpler option is to use one of the board defintions and "patch" the pins_arduino.h before a build. I have done that for the s3 board since the defines for the RGB was faulty. 
>>
>>See; script/patch_arduino.py for an example. 
>>
>>-- 
>>Reply to this email directly or view it on GitHub:
>>https://github.com/mp-se/gravitymon/discussions/190#discussioncomment-11709796
>>You are receiving this because you were mentioned.
>>
>>Message ID: ***@***.***>
>-- 
>Sent from my Android phone with K-9 Mail. Please excuse my brevity.
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

**Reply by mp-se** [reply]
Please open a pull request with the changes in the code and documentation. Please publish your hardware releated documentation and pcb information on a separate site/github repository which can be linked from the documentation.

**Comment by dtashmore547** [comment]
ok Magnus
i will try, not done it before but always a first time, not too many changes as the files take care of most differences. hope the is no way i can mess up.
regards
Dave

On 7 January 2025 19:07:00 GMT+02:00, Magnus ***@***.***> wrote:
>Please open a pull request with the changes in the code and documentation. Please publish your hardware releated documentation and pcb information on a separate site/github repository which can be linked from the documentation. 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/discussions/190#discussioncomment-11763507
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

**Reply by mp-se** [reply]
I will validate the changes before I merge it anyway. I suggest you create a separate branch for the pull request since all changes you do there will be reflected in the pull request. This way you can do changes if there is a need and update the PR

**Comment by dtashmore547** [comment]
hi Magnus
not having any luck with the pull request, probably don't have enough understanding. it tells me the pull request has failed i must be a collaborator. 
regards
Dave

On 7 January 2025 19:07:00 GMT+02:00, Magnus ***@***.***> wrote:
>Please open a pull request with the changes in the code and documentation. Please publish your hardware releated documentation and pcb information on a separate site/github repository which can be linked from the documentation. 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/discussions/190#discussioncomment-11763507
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

**Reply by mp-se** [reply]
You need to push your changes to github and your fork before you can create a pull request from your fork on github. You cant push directly to my master

**Comment by xdigitall** [comment]
> I have updated my super mini boards to the current dev version and everything is working fine, I still have a desire to see the led in operation as it gives a useful indication of the state of the device. I am lost as to why the led should interfere with the gyro as I have the gyro on gpio6 and gpio7 whereas the blue on-board LED is on gpio8. I would love to see the LED operating the same as the on-board LED of the esp8266 mini. is this possible? could I define the super mini as a separate device place this definition in all the places where the esp32c3 definition is used except the led.hpp and led.cpp and place the gpio8 as LED_INBUILT in the main.hpp folder or would that cause conflicts?

Hello! I would like also to make a hydrometer based on the TENSTAR ROBOT ESP32-C3 SuperMini microcontroller, and I am happy to hear that you have already successfully adapted gravitymont to it. Could I ask for help with what kind of wiring diagram you used to assemble the hardware? Thank you!

**Reply by mp-se** [reply]
This is a user contribution so i dont support it myself. 

The author of this thread should be able to answer your questions or you need to dig into the board pin configurations and start there

**Reply by dtashmore547** [reply]
hi xdigitall
presuming the ten star robot version is the same as my generic version then the adaptation is quite easy, Magnus created the software environment for it. 
i have been setting up a project page on pcbway.com for the board i made. 
https://www.pcbway.com/project/shareproject/W58186ASV15_W58186ASV12_GravityMon_1b172f03.html
it is not quite complete and i need to make some corrections but should be of assistance to you, i have the circuitry there plus some instructions.
one thing i have found on the generic esp32 c3 super mini is that the onboard regulator isn't suitable and drains the battery it also has a high dropout voltage, so i removed it on my boards and bridged from the regulator on the gyro which is more suitable to low power operation.

**Reply by xdigitall** [reply]
Thank you for the quick response! Very thorough and useful work!

**Comment by dtashmore547** [comment]
thanks, if you experience any problems you may contact me, i still have to update the project page, i am currently looking at the endurance which i can only consider as fair.

**Comment by mp-se** [comment]
I did see that the board you added @dtashmore547 was missing the bluetooth option which could be the reason for BLE not working. I added that last week so hopefully the BLE features should work now. I dont have any boards so I cant test the release.

**Comment by mp-se** [comment]
I saw that there are multiple boards that have the name super mini and some of them have a different pin configuration. I will rename your config and target to tenstar to make it clear which board the target is supporting

**Comment by mp-se** [comment]
I had a look at your arduino_pins.h and that does not seam to match the tenstar board, not all the pins are included... pins 10-13 are not included in the header definition. This might be for another super mini manufacturer.

**Comment by dtashmore547** [comment]
hi Magnus
had a look at the ten star board when the issue came up and it is visually identical to the ones i use, never noticed any reference to pins 11,12 and 13 but pin 10 is there. all the boards i have purchased under that moniker have been identical, can you please send me any reference to super mini boards with a different pinout.
greatest regards
Dave

On 6 June 2025 22:32:12 GMT+02:00, Magnus ***@***.***> wrote:
>I had a look at your arduino_pins.h and that does not seam to match the tenstar board, not all the pins are included... pins 10-13 are not included in the header definition. This might be for another super mini manufacturer.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/discussions/190#discussioncomment-13393926
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

**Comment by mp-se** [comment]
This is the tenstar board i found on aliexpress and you have pins 10-13 on the right side.

<img width="306" alt="image" src="https://github.com/user-attachments/assets/a139b3d8-1134-4104-b71f-de4898962900" />

I also searched for ESP32c3 super mini boards on amazon here in sweden and found 5 boards that where named super mini, none of them had the same pin configuration and it also varied from 14 to 22 pins... So therefore i have changed the configuration name of your contribution to tenstar_esp32c3_super_mini but I also noticed that the 10-13 pins are not defined in the pins_arduino.h so I'm wondering if that is the correct config for that board... super mini variants seams to be very much manufacturer specific.

**Comment by dtashmore547** [comment]
the board on the front is the S3 version, you can tell by the larger chip, i did buy one of them and they have a completely different pin configuration. the C3 chip is noticeably smaller.

On 7 June 2025 11:11:41 GMT+02:00, Magnus ***@***.***> wrote:
>This is the tenstar board i found on aliexpress and you have pins 10-13 on the right side.
>
><img width="306" alt="image" src="https://github.com/user-attachments/assets/a139b3d8-1134-4104-b71f-de4898962900" />
>
>I also searched for ESP32c3 super mini boards on amazon here in sweden and found 5 boards that where named super mini, none of them had the same pin configuration and it also varied from 14 to 22 pins... So therefore i have changed the configuration name of your contribution to tenstar_esp32c3_super_mini but I also noticed that the 10-13 pins are not defined in the pins_arduino.h so I'm wondering if that is the correct config for that board... super mini variants seams to be very much manufacturer specific. 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/discussions/190#discussioncomment-13396470
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

**Reply by mp-se** [reply]
Well, thats another problem if they have the wrong picture on their listning...

**Comment by dtashmore547** [comment]
looking at the photo it seems to me that they were advertising 4 different boards, probably just have to be careful to buy the correct one. i see this type of advert regularly on AliExpress.

On 7 June 2025 12:09:35 GMT+02:00, Magnus ***@***.***> wrote:
>Well, thats another problem if they have the wrong picture on their listning...
>
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/discussions/190#discussioncomment-13396693
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

**Comment by mp-se** [comment]
Ok, i did a new search and that seams to be almost correct, the header file is missing GPIO10 other than that it looks correct.
